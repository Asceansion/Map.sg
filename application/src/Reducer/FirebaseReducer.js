import { SIGN_OUT_SUCCESS, SIGN_IN_SUCCESS, SIGN_UP_SUCCESS, SIGN_UP_FAILURE, SIGN_IN_FAILURE, SIGN_OUT_FAILURE, CLEAR_ERROR_MESSAGE } from '../Constants/actionTypes';

/**
 * Home reducers to update states belonging to Home view
 * 
 * @Koh Tong Liang
 * @Version 1
 * @Since 31/10/2020
*/

const initialState = {
    loggedIn: false,
    user: null,
    errorMessage: '',
}

function FirebaseReducer (state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case SIGN_IN_SUCCESS:
            return Object.assign({}, state, {
                loggedIn: true,
                user: action.data,
            });
        case SIGN_OUT_SUCCESS:
            return initialState;
        case SIGN_UP_SUCCESS:
            return Object.assign({}, state, {
                loggedIn: true,
                user: action.data,
            });
        case SIGN_IN_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        case SIGN_OUT_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        case SIGN_UP_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        case CLEAR_ERROR_MESSAGE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage
            });
        default:
    } // end of switch case

    return state;
}

export default FirebaseReducer;