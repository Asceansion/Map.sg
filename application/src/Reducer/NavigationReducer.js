import { SEARCH_START_LOCATION_SUCCEEDED, SEARCH_END_LOCATION_SUCCEEDED, PROCESS_START_LOCATION,
    PROCESS_END_LOCATION, PLAN_ROUTE_SUCCEEDED, TRIP_SUMMARY, SAVE_TRIP, MAP_MATCHING_SUCCEEDED, CANCEL_ROUTE, UPDATE_STEPS } from '../Constants/actionTypes';

/**
 * Home reducers to update states belonging to Home view
 * @Koh Tong Liang
 * @Version 1
 * @Since 31/10/2020
*/

const initialState = {
    startLocation: [],
    endLocation: [],
    startLocationSearchResult: [], // for routeplanner form
    endLocationSearchResult: [], // for routeplanner form
    navigationRoute: [], // geometric data of the route user is taking
    mapMatchedRoute: [], // fixed route linestring
    tripSummaryView: false, // toggle view for end of trip summary
    onRoute: false, // show that user is on journey
    stepNo: 0, // indiciate which stage of step by step instruction user is in
    routeInstruction: [], // stores step by step instruction
}

function NavigationReducer (state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case UPDATE_STEPS:
            return Object.assign({}, state, {
                stepNo: initialState.stepNo + action.payload,
            });
        case SEARCH_START_LOCATION_SUCCEEDED:
            return Object.assign({}, state, {
                startLocationSearchResult: state.startLocationSearchResult.concat(action.payload)
            });
        case SEARCH_END_LOCATION_SUCCEEDED:
            return Object.assign({}, state, {
                endLocationSearchResult: state.endLocationSearchResult.concat(action.payload)
            });
        case PROCESS_START_LOCATION:
            return Object.assign({}, state, {
                startLocationSearchResult: initialState.startLocationSearchResult,
                startLocation: state.startLocation.concat({lng: action.payload.geometry.coordinates[0], lat: action.payload.geometry.coordinates[1]}),
            });
        case PROCESS_END_LOCATION:
            return Object.assign({}, state, {
                endLocationSearchResult: initialState.endLocationSearchResult,
                endLocation: state.endLocation.concat({lng: action.payload.geometry.coordinates[0], lat: action.payload.geometry.coordinates[1]}),
            });
        case PLAN_ROUTE_SUCCEEDED:
            return Object.assign({}, state, {
                navigationRoute: state.navigationRoute.concat(action.payload.route),
                onRoute: !state.onRoute,
                routeInstruction: state.routeInstruction.concat(action.payload.routeInstruction),
            });
        case MAP_MATCHING_SUCCEEDED:
            return Object.assign({}, state, {
                mapMatchedRoute: state.mapMatchedRoute.concat(action.payload),
            });
        case TRIP_SUMMARY:
            return Object.assign({}, state, {
                ...initialState,
                tripSummaryView: !state.tripSummaryView
            });
        case CANCEL_ROUTE:
            // return initialState;
            return Object.assign({}, state, {
                ...initialState,
                navigationRoute: initialState.navigationRoute,
            });
        case SAVE_TRIP:
            // store navigation route
            break;
        default:
    } // end of switch case
    return state;
}

export default NavigationReducer;