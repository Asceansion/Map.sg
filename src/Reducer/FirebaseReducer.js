// import action types
import {
    SIGN_OUT_SUCCESS, SIGN_IN_SUCCESS, SIGN_UP_SUCCESS, SIGN_UP_FAILURE, SIGN_IN_FAILURE,
    SIGN_OUT_FAILURE, CLEAR_ERROR_MESSAGE, SAVE_HISTORY_SUCCESS, LOAD_HISTORY_SUCCESS
} from '../Constants/actionTypes';

// Initialise initial values for state
const initialState = {
    loggedIn: false,
    user: null,
    errorMessage: '',
    signInSuccess: false,
    signUpSuccess: false,
    history: [],
}

/**
 * Firebase reducers to update states pertaining to states related to user accounts and data storage
 * @author Koh Tong Liang
 * @version 1
 * @since 31/10/2020
*/
function FirebaseReducer(state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case SAVE_HISTORY_SUCCESS:
            // return Object.assign({}, state, {
            //     history: state.history.concat(action.doc)
            // });
            return;
        case LOAD_HISTORY_SUCCESS:
            return Object.assign({}, state, {
                history: initialState.history.concat(action.payload),
            });
        case SIGN_IN_SUCCESS:
            return Object.assign({}, state, {
                loggedIn: true,
                signInSuccess: !state.signInSuccess,
                user: action.data,
            });
        case SIGN_OUT_SUCCESS:
            return initialState;
        case SIGN_UP_SUCCESS:
            return Object.assign({}, state, {
                loggedIn: true,
                signUpSuccess: !state.signUpSuccess,
                user: action.data,
            });
        case SIGN_IN_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        case SIGN_OUT_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        case SIGN_UP_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        default:
    } // end of switch case

    return state;
}

export default FirebaseReducer;