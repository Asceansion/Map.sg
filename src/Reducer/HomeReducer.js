import { GET_USER_LOCATION_SUCCEEDED, HISTORY_VIEW, TOGGLE_ROUTE_PLANNER,
    TOGGLE_TRAFFIC_IMAGES_VIEW, TOGGLE_MAP_PICKER, RETURN_MAP_PICKER_RESULT,
    BOOKMARK_VIEW } from '../Constants/actionTypes';
/**
 * Home reducers to update states belonging to Home view
 * 
 * @Koh Tong Liang
 * @Version 1
 * @Since 31/10/2020
*/

const initialState = {
    routePlannerView: false,
    trafficImagesView: false,
    userLocation: [],
    historyView: false,
    bookmarkView: false,
    mapPickerMode: false,
    mapPickerResult: [],
}

function HomeReducer (state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case 'OVERRIDE_USER_LOCATION':
            return Object.assign({}, state, {
                userLocation: initialState.userLocation.concat({lng: action.payload.lng, lat: action.payload.lat})
            });
        case GET_USER_LOCATION_SUCCEEDED:
            console.log(action.payload)
            return Object.assign({}, state, {
                userLocation: initialState.userLocation.concat(action.payload)
            });
        case TOGGLE_ROUTE_PLANNER:
            return Object.assign({}, state, {
                routePlannerView: !state.routePlannerView
            });
        case TOGGLE_MAP_PICKER:
            return Object.assign({}, state, {
                mapPickerMode: !state.mapPickerMode
            });
        case HISTORY_VIEW:        
            return Object.assign({}, state, {
                historyView: !state.historyView
            });
        case BOOKMARK_VIEW:
            return Object.assign({}, state, {
                bookmarkView: !state.bookmarkView
            });
        case RETURN_MAP_PICKER_RESULT:
            return Object.assign({}, state, {
                mapPickerResult: initialState.mapPickerResult.concat(action.payload),
            });
        case TOGGLE_TRAFFIC_IMAGES_VIEW:
            break;
        default:
    } // end of switch case

    return state;
}

export default HomeReducer;