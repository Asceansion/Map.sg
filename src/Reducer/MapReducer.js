// import action types
import {
    GET_TRAFFIC_IMAGES_SUCCEEDED, GET_ERP_DATA_SUCCEEDED,
    UPDATE_CAMERA_MARKERS, UPDATE_LINE_STRING, UPDATE_NEXT_CAMERA, SET_USER_MARKER, SET_PINNED_CAMERA_MARKERS, SET_ERP_MARKERS, SET_STEP_MARKERS, SET_MAP
} from '../Constants/actionTypes';

// Initialise initial values for state
const initialState = {
    cameras: [],
    cameraMarkers: [],
    ERP: [],
    lineString: [],
    userMarker: [],
    pinnedCameraMarkers: [],
    erpMarkers: [],
    stepMarkers: [],
    map: [],
}



/**
 * Map Reducer serves to update the states pertaining to the map view. Such as camera and erp positions and list
 * @author Jeremiah
 * @version 1
 * @since 30/10/2020
*/
export default function MapReducer(state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case UPDATE_CAMERA_MARKERS:
            return Object.assign({}, state, {
                cameraMarkers: initialState.cameraMarkers.concat(action.payload)
            });
        case GET_TRAFFIC_IMAGES_SUCCEEDED:
            return Object.assign({}, state, {
                cameras: initialState.cameras.concat(action.payload)
            });
        case GET_ERP_DATA_SUCCEEDED:
            return Object.assign({}, state, {
                ERP: initialState.ERP.concat(action.payload)
            });
        case UPDATE_LINE_STRING:
            return Object.assign({}, state, {
                lineString: initialState.lineString.concat(action.payload)
            });
        case UPDATE_NEXT_CAMERA:
            return Object.assign({}, state, {
                cameraMarkers: initialState.cameraMarkers.concat(action.payload)
            });
        case SET_USER_MARKER:
            return Object.assign({}, state, {
                userMarker: initialState.userMarker.concat(action.payload)
            });
        case SET_PINNED_CAMERA_MARKERS:
            return Object.assign({}, state, {
                pinnedCameraMarkers: initialState.pinnedCameraMarkers.concat(action.payload)
            });
        case SET_ERP_MARKERS:
            return Object.assign({}, state, {
                erpMarkers: initialState.erpMarkers.concat(action.payload)
            });
        case SET_STEP_MARKERS:
            return Object.assign({}, state, {
                stepMarkers: initialState.stepMarkers.concat(action.payload)
            });
        case SET_MAP:
            return Object.assign({}, state, {
                map: initialState.map.concat(action.payload)
            });
        default:
    }

    return state;
}