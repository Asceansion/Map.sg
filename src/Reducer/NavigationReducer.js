import {
    SEARCH_START_LOCATION_SUCCEEDED, SEARCH_END_LOCATION_SUCCEEDED, PROCESS_START_LOCATION,
    PROCESS_END_LOCATION, PLAN_ROUTE_SUCCEEDED, TRIP_SUMMARY, MAP_MATCHING_SUCCEEDED,
    CANCEL_ROUTE, UPDATE_STEPS, REROUTE_SUCCEEDED, FILTER_ROUTE_ERP, SAVE_ROUTE_NAME,
    RUN_HISTORY, SET_START_LOCATION_SEARCH, SET_END_LOCATION_SEARCH,TOGGLE_ROUTE_PLANNER,
    TOGGLE_MAP_PICKER, RETURN_MAP_PICKER_RESULT
} from '../Constants/actionTypes';

// Initialise initial values for state
const initialState = {
    routePlannerView: false,
    startLocation: [],
    endLocation: [],
    routeName: [],
    startLocationSearchResult: [], // for routeplanner form
    endLocationSearchResult: [], // for routeplanner form
    navigationRoute: [], // geometric data of the route user is taking
    mapMatchedRoute: [], // fixed route linestring
    tripSummaryView: false, // toggle view for end of trip summary
    onRoute: false, // show that user is on journey
    stepNo: 0, // indiciate which stage of step by step instruction user is in
    routeInstruction: [], // stores step by step instruction
    erpCharge: [], // store erp charges along the
    erpTotalCharge: [], // store total erp charges incurred
    erpFiltered: [],
    startLocationSearch: '',
    endLocationSearch: '',
    mapPickerMode: 0,
    mapPickerResult: [],
}

/**
 * Navigation reducers to update states pertaining to navigation such as route information, start/end locations
 * @Koh Tong Liang
 * @Version 1
 * @Since 31/10/2020
*/
function NavigationReducer(state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case UPDATE_STEPS:
            return Object.assign({}, state, {
                stepNo: initialState.stepNo + action.payload,
            });
        case TOGGLE_MAP_PICKER:
            return Object.assign({}, state, {
                mapPickerMode: initialState.mapPickerMode + action.payload,
                routePlannerView: !state.routePlannerView,
            });
        case TOGGLE_ROUTE_PLANNER:
            return Object.assign({}, state, {
                startLocationSearch: initialState.startLocationSearch,
                endLocationSearch: initialState.endLocationSearch,
                routePlannerView: !state.routePlannerView,
            });
        case SET_START_LOCATION_SEARCH:
            return Object.assign({}, state, {
                startLocationSearch: initialState.startLocationSearch.concat(action.payload),
            });
        case SET_END_LOCATION_SEARCH:
            return Object.assign({}, state, {
                endLocationSearch: initialState.endLocationSearch.concat(action.payload),
            });
        case RETURN_MAP_PICKER_RESULT:
            if (state.mapPickerMode == 1) {
                return Object.assign({}, state, {
                    mapPickerResult: initialState.mapPickerResult.concat(action.payload),
                    startLocation: initialState.startLocation.concat(action.payload),
                    startLocationSearch: initialState.startLocationSearch.concat('Pinned location'),
                    mapPickerMode: initialState.mapPickerMode,
                    routePlannerView: !state.routePlannerView,
                });
            } else if (state.mapPickerMode == 2) {
                return Object.assign({}, state, {
                    mapPickerResult: initialState.mapPickerResult.concat(action.payload),
                    endLocation: initialState.endLocation.concat(action.payload),
                    endLocationSearch: initialState.endLocationSearch.concat('Pinned location'),
                    mapPickerMode: initialState.mapPickerMode,
                    routePlannerView: !state.routePlannerView,
                });
            }
        case SEARCH_START_LOCATION_SUCCEEDED:
            return Object.assign({}, state, {
                startLocationSearchResult: initialState.startLocationSearchResult.concat(action.payload)
            });
        case SEARCH_END_LOCATION_SUCCEEDED:
            return Object.assign({}, state, {
                endLocationSearchResult: initialState.endLocationSearchResult.concat(action.payload)
            });
        case PROCESS_START_LOCATION:
            return Object.assign({}, state, {
                startLocationSearchResult: initialState.startLocationSearchResult,
                startLocation: initialState.startLocation.concat(action.payload),
            });
        case PROCESS_END_LOCATION:
            return Object.assign({}, state, {
                endLocationSearchResult: initialState.endLocationSearchResult,
                endLocation: initialState.endLocation.concat(action.payload),
            });
        case PLAN_ROUTE_SUCCEEDED:
            return Object.assign({}, state, {
                navigationRoute: initialState.navigationRoute.concat(action.payload.route),
                onRoute: !initialState.onRoute,
                routeInstruction: initialState.routeInstruction.concat(action.payload.routeInstruction),
            });
        case REROUTE_SUCCEEDED:
            return Object.assign({}, state, {
                stepNo: initialState.stepNo,
                onRoute: !initialState.onRoute,
                navigationRoute: initialState.navigationRoute.concat(action.payload.route),
                routeInstruction: initialState.routeInstruction.concat(action.payload.routeInstruction),
            });
        case MAP_MATCHING_SUCCEEDED:
            return Object.assign({}, state, {
                mapMatchedRoute: state.mapMatchedRoute.concat(action.payload),
            });
        case TRIP_SUMMARY:
            return Object.assign({}, state, {
                ...initialState,
                tripSummaryView: !state.tripSummaryView
            });
        case RUN_HISTORY:
            return Object.assign({}, state, {
                navigationRoute: initialState.navigationRoute.concat(action.payload.routeInformation),
                onRoute: !initialState.onRoute,
                routeInstruction: initialState.routeInstruction.concat(action.payload.routeInformation.data.routes[0].legs[0].steps),
                startLocation: initialState.startLocation.concat(action.payload.startLocation),
                endLocation: initialState.endLocation.concat(action.payload.endLocation),
                routeName: initialState.routeName.concat(action.payload.routeName),
            });
        case CANCEL_ROUTE:
            return Object.assign({}, state, {
                ...initialState,
                navigationRoute: initialState.navigationRoute,
            });
        case FILTER_ROUTE_ERP:
            return Object.assign({}, state, {
                erpFiltered: initialState.erpFiltered.concat(action.payload)
            });
        case SAVE_ROUTE_NAME:
            return Object.assign({}, state, {
                routeName: initialState.routeName.concat(action.payload)
            });
        default:
    } // end of switch case
    return state;
}

export default NavigationReducer;