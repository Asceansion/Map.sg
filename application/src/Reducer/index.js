import { combineReducers } from 'redux';
import HomeReducer from './HomeReducer';
import NavigationReducer from './NavigationReducer';

/* *
 * This is the root reducers, it uses combineReducers to group multiple reducers together
 * and export all of them at 1 go. This serves as entry point to all the other reducers.
 * 
 * @Koh Tong Liang
 * @Version 1.0
 * @Since 30/10/2020
 * */

export default combineReducers({ HomeReducer, NavigationReducer });
