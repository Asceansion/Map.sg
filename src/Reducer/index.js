import { combineReducers } from 'redux';
import HomeReducer from './HomeReducer';
import NavigationReducer from './NavigationReducer';
import MapReducer from './MapReducer';
import FirebaseReducer from './FirebaseReducer';

/* *
 * This is the root reducers, it uses combineReducers to group multiple reducers together
 * and export all of them at 1 go. This serves as entry point to all the other reducers.
 * @author Koh Tong Liang
 * @version 1.0
 * @since 30/10/2020
 * */

export default combineReducers({ 
    HomeReducer,
    NavigationReducer,
    MapReducer,
    FirebaseReducer,
});
