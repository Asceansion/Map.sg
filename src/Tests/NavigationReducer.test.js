import NavigationReducer from '../Reducer/NavigationReducer'
import {NavigationState} from '../Reducer/NavigationReducer'




describe('NavigationReducer ', () => {
    test('should return the initial state', () => {
      expect(NavigationReducer(undefined, {})).toMatchSnapshot()
    })
  
    test('should handle UPDATE_STEPS', () => {
      expect(
        NavigationReducer(NavigationState,
        {
          type: 'UPDATE_STEPS'
        })
      ).toMatchSnapshot()
    })
  
    test('should handle SEARCH_START_LOCATION_SUCCEEDED', () => {
      expect(
        NavigationReducer(NavigationState,
        {
          type: 'SEARCH_START_LOCATION_SUCCEEDED'
        })
      ).toMatchSnapshot()
    })
  
    test('should handle SEARCH_END_LOCATION_SUCCEEDED', () => {
      expect(
        NavigationReducer(NavigationState,
        {
          type: 'SEARCH_END_LOCATION_SUCCEEDED'
        })
      ).toMatchSnapshot()
    })
    test('should handle PROCESS_START_LOCATION', () => {
        expect(
          NavigationReducer(NavigationState,
          {
            type: 'PROCESS_START_LOCATION'
          })
        ).toMatchSnapshot()
      })
      test('should handle PROCESS_END_LOCATION', () => {
        expect(
          NavigationReducer(NavigationState,
          {
            type: 'PROCESS_END_LOCATION'
          })
        ).toMatchSnapshot()
      })
      test('should handle REROUTE_SUCCEEDED', () => {
        expect(
          NavigationReducer(NavigationState,
          {
            type: 'REROUTE_SUCCEEDED'
          })
        ).toMatchSnapshot()
      })
      test('should handle MAP_MATCHING_SUCCEEDED', () => {
        expect(
          NavigationReducer(NavigationState,
          {
            type: 'MAP_MATCHING_SUCCEEDED'
          })
        ).toMatchSnapshot()
      })
      test('should handle TRIP_SUMMARY', () => {
        expect(
          NavigationReducer(NavigationState,
          {
            type: 'TRIP_SUMMARY'
          })
        ).toMatchSnapshot()
      })
      test('should handle CANCEL_ROUTE', () => {
        expect(
          NavigationReducer(NavigationState,
          {
            type: 'CANCEL_ROUTE'
          })
        ).toMatchSnapshot()
      })
      test('should handle FILTER_ROUTE_ERP', () => {
        expect(
          NavigationReducer(NavigationState,
          {
            type: 'FILTER_ROUTE_ERP'
          })
        ).toMatchSnapshot()
      })
      test('should handle SAVE_ROUTE_NAME', () => {
        expect(
          NavigationReducer(NavigationState,
          {
            type: 'SAVE_ROUTE_NAME'
          })
        ).toMatchSnapshot()
      })
  })
