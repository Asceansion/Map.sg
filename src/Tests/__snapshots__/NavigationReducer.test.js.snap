// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationReducer  should handle CANCEL_ROUTE 1`] = `
Object {
  "endLocation": Array [],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [],
  "erpCharge": Array [],
  "erpFiltered": Array [],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [],
  "routePlannerView": false,
  "startLocation": Array [],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [],
  "stepNo": 0,
  "tripSummaryView": false,
}
`;

exports[`NavigationReducer  should handle FILTER_ROUTE_ERP 1`] = `
Object {
  "endLocation": Array [],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [],
  "erpCharge": Array [],
  "erpFiltered": Array [
    undefined,
  ],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [],
  "routePlannerView": false,
  "startLocation": Array [],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [],
  "stepNo": 0,
  "tripSummaryView": false,
}
`;

exports[`NavigationReducer  should handle MAP_MATCHING_SUCCEEDED 1`] = `
Object {
  "endLocation": Array [],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [],
  "erpCharge": Array [],
  "erpFiltered": Array [],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [
    undefined,
  ],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [],
  "routePlannerView": false,
  "startLocation": Array [],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [],
  "stepNo": 0,
  "tripSummaryView": false,
}
`;

exports[`NavigationReducer  should handle PROCESS_END_LOCATION 1`] = `
Object {
  "endLocation": Array [
    undefined,
  ],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [],
  "erpCharge": Array [],
  "erpFiltered": Array [],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [],
  "routePlannerView": false,
  "startLocation": Array [],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [],
  "stepNo": 0,
  "tripSummaryView": false,
}
`;

exports[`NavigationReducer  should handle PROCESS_START_LOCATION 1`] = `
Object {
  "endLocation": Array [],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [],
  "erpCharge": Array [],
  "erpFiltered": Array [],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [],
  "routePlannerView": false,
  "startLocation": Array [
    undefined,
  ],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [],
  "stepNo": 0,
  "tripSummaryView": false,
}
`;

exports[`NavigationReducer  should handle REROUTE_SUCCEEDED 1`] = `
Object {
  "endLocation": Array [],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [],
  "erpCharge": Array [],
  "erpFiltered": Array [],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [],
  "routePlannerView": false,
  "startLocation": Array [],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [],
  "stepNo": 0,
  "tripSummaryView": false,
}
`;

exports[`NavigationReducer  should handle SAVE_ROUTE_NAME 1`] = `
Object {
  "endLocation": Array [],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [],
  "erpCharge": Array [],
  "erpFiltered": Array [],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [
    undefined,
  ],
  "routePlannerView": false,
  "startLocation": Array [],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [],
  "stepNo": 0,
  "tripSummaryView": false,
}
`;

exports[`NavigationReducer  should handle SEARCH_END_LOCATION_SUCCEEDED 1`] = `
Object {
  "endLocation": Array [],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [
    undefined,
  ],
  "erpCharge": Array [],
  "erpFiltered": Array [],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [],
  "routePlannerView": false,
  "startLocation": Array [],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [],
  "stepNo": 0,
  "tripSummaryView": false,
}
`;

exports[`NavigationReducer  should handle SEARCH_START_LOCATION_SUCCEEDED 1`] = `
Object {
  "endLocation": Array [],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [],
  "erpCharge": Array [],
  "erpFiltered": Array [],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [],
  "routePlannerView": false,
  "startLocation": Array [],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [
    undefined,
  ],
  "stepNo": 0,
  "tripSummaryView": false,
}
`;

exports[`NavigationReducer  should handle TRIP_SUMMARY 1`] = `
Object {
  "endLocation": Array [],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [],
  "erpCharge": Array [],
  "erpFiltered": Array [],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [],
  "routePlannerView": false,
  "startLocation": Array [],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [],
  "stepNo": 0,
  "tripSummaryView": true,
}
`;

exports[`NavigationReducer  should handle UPDATE_STEPS 1`] = `
Object {
  "endLocation": Array [],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [],
  "erpCharge": Array [],
  "erpFiltered": Array [],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [],
  "routePlannerView": false,
  "startLocation": Array [],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [],
  "stepNo": NaN,
  "tripSummaryView": false,
}
`;

exports[`NavigationReducer  should return the initial state 1`] = `
Object {
  "endLocation": Array [],
  "endLocationSearch": "",
  "endLocationSearchResult": Array [],
  "erpCharge": Array [],
  "erpFiltered": Array [],
  "erpTotalCharge": Array [],
  "mapMatchedRoute": Array [],
  "mapPickerMode": 0,
  "mapPickerResult": Array [],
  "navigationRoute": Array [],
  "onRoute": false,
  "routeInstruction": Array [],
  "routeName": Array [],
  "routePlannerView": false,
  "startLocation": Array [],
  "startLocationSearch": "",
  "startLocationSearchResult": Array [],
  "stepNo": 0,
  "tripSummaryView": false,
}
`;
